etl/steps/data/garden/regions/2023-01-01/regions.py

```diff
@@ -0,0 +1,149 @@
+"""Region definitions, aliases, members, historical transitions and codes.
+
+This step will load the adjacent files of region definitions and codes, and create a garden dataset with tables:
+* "definitions": Region definitions (name, country type, etc.).
+* "aliases": Region aliases (i.e. variants of the region name).
+* "members": Region members (roughly, sub-regions that would need to be added up when aggregating data for the region).
+* "related": Other possible region members to be aware of (e.g. regions with an unclear official status that may be
+  members of the region according to some data providers).
+* "transitions": Historical transitions between regions.
+* "legacy_codes": Legacy codes (e.g. wikidata code).
+
+"""
+
+import pandas as pd
+import yaml
+from owid.catalog import Table
+
+from etl.helpers import PathFinder, create_dataset
+
+# Get paths and naming conventions for current step.
+paths = PathFinder(__file__)
+
+# Path to input file of region definitions.
+REGION_DEFINITIONS_FILE = paths.directory / "regions.yml"
+# Path to input file of region codes.
+REGION_CODES_FILE = paths.directory / "regions.codes.csv"
+
+# Expected region types.
+REGION_TYPES = ["country", "continent", "aggregate", "other"]
+
+
+def parse_raw_definitions(df: pd.DataFrame) -> pd.DataFrame:
+    # If "short_name" is not given, fill with "name".
+    df["short_name"] = df["short_name"].fillna(df["name"])
+
+    # If "region_type" is not given, fill with "country" (the most common region type).
+    df["region_type"] = df["region_type"].fillna("country")
+
+    # If "defined_by" is not given, fill with "owid".
+    if "defined_by" not in df.columns:
+        df["defined_by"] = pd.Series(dtype=str)
+    df["defined_by"] = df["defined_by"].fillna("owid")
+
+    # If "is_historical" is not given, fill with False.
+    df["is_historical"] = df["is_historical"].fillna(False)
+
+    return df
+
+
+def run_sanity_checks(df: pd.DataFrame) -> None:
```

Comment:
would place it after method `parse_raw_definitions`, after `run`.

Reply:
I usually define `run` at the very end, and all additional functions are defined before `run`, and tend to appear in order of appearance in `run`. What's your criterion?